### DISCUSSION TITLE
angular/what-is-the-difference-between-constructor-and-ngoninit-in-angular/

### DISCUSSION THREAD

**Discussion by giscus** [arungudelli/angularjswiki#164] [Announcements]
# angular/what-is-the-difference-between-constructor-and-ngoninit-in-angular/

ngOnInit and constructor methods in Angular might be confusing to you,if you are new to Angular.In this tutorial, we will cover the differences between ngOnInit & constructor.

https://www.angularjswiki.com/angular/what-is-the-difference-between-constructor-and-ngoninit-in-angular/

**Comment by bharukarupesh11** [comment]
Hello, I've tried doing the same in my project as shown in the above example. **However, I'm still getting undefined in the ngOnInit( ) method of my child component. Any help is really appreciated. The only difference in my code is, I'm passing async data from the parent component to my child component.** I think, that might be causing an issue.

**Reply by arungudelli** [reply]
Do you have any demo link stackblitz or something so that i can have a look

**Reply by bharukarupesh11** [reply]
Could you kindly refer to this stackoverflow link: https://stackoverflow.com/a/70798686/15684466

**Reply by arungudelli** [reply]
Yes you should use ngOnChanges method

**Reply by bharukarupesh11** [reply]
**So, you mean, the approach that I've used is correct?** Like, instantiating an empty object in parent component like `customer:Customer = new Customer( );` and then using ngOnChanges( ) method inside the child class.

**Reply by arungudelli** [reply]
I feel that's the way.. 

Go through the article to understand it further. 
https://angular.io/guide/lifecycle-hooks#onchanges

**Reply by bharukarupesh11** [reply]
> I feel that's the way..
> 
> Go through the article to understand it further. https://angular.io/guide/lifecycle-hooks#onchanges

**Thanks @arungudelli for your kind response** 👍 :), I'll check the article.

**Reply by arungudelli** [reply]
Please star the repo.. it it helps.. Thanks in advance..

**Reply by bharukarupesh11** [reply]
> Please star the repo.. it it helps.. Thanks in advance..

**Done.**
